src/main/java/com/uid2/admin/vertx/service/ServiceService.java

```diff
@@ -169,26 +176,72 @@ private void handleServiceRoles(RoutingContext rc) {
                 return;
             }
 
-            final Set<Role> roles;
-            try {
-                roles = rolesSpec.stream().map(s -> Role.valueOf((String) s)).collect(Collectors.toSet());
-            } catch (IllegalArgumentException e) {
-                ResponseUtil.error(rc, 400, "invalid parameter: roles");
-                return;
+            if (rolesSpec != null) {
+                final Set<Role> roles;
+                try {
+                    roles = rolesSpec.stream().map(s -> Role.valueOf((String) s)).collect(Collectors.toSet());
+                } catch (IllegalArgumentException e) {
+                    ResponseUtil.error(rc, 400, "invalid parameter: roles");
+                    return;
+                }
+                service.setRoles(roles);
```

Comment:
is `roles` guaranteed to be non-empty here?

Reply:
We allow it to be non empty in other places like the client key. But if we are able to delete these then maybe it is a good idea to not allow roles to be non-empty, and instead delete the service if it can't do anything.

Reply:
Updated the validation of the roles and added some tests. We don't actually use roles on the service at the moment for actually validating what the service can do, so this is a future enhancement

Reply:
Oh I didn't know that. We should create a ticket to track this.

Reply:
Ticket created: UID2-2084